### PR TITLE
divide, ln, log, mod: Clarified behavior for 0 input / infinity results

### PR DESCRIPTION
**Pull request by m-mohr**
We mention NaN here the first time, I think.

Related: #468, #472, #474

### CODE REVIEW

File: divide.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
     "id": "divide",
     "summary": "Division of two numbers",
-    "description": "Divides argument `x` by the argument `y` (*`x / y`*) and returns the computed result.\n\nNo-data values are taken into account so that `null` is returned if any element is such a value.\n\nThe computations follow [IEEE Standard 754](https://ieeexplore.ieee.org/document/8766229) whenever the processing environment supports it. Therefore, a division by zero results in ±infinity if the processing environment supports it. Otherwise, a `DivisionByZero` exception must the thrown.",
+    "description": "Divides argument `x` by the argument `y` (*`x / y`*) and returns the computed result.\n\nNo-data values are taken into account so that `null` is returned if any element is such a value.\n\nThe computations follow [IEEE Standard 754](https://ieeexplore.ieee.org/document/8766229) whenever the processing environment supports it. A division by zero results in:\n\n- +infinity for `x` > 0,\n- -infinity for `x` < 0,\n- `NaN` for `x` = 0,\n- or otherwise, throws a `DivisionByZero` exception if the other options are not supported by the processing environment.",
```
Comment (soxofaan) [inline divide.json:4]: > +infinity for `x` > 0,\n- -infinity for `x` < 0,\n-

I'm not sure this is correct (I'm also not sure how tightened down the spec should be here): 
the sign of infinity also depends on which side you're looking from, e.g. `1/y` at `y=0` is +infinity  if you "approach" that limit it from positive y values, but -infinity if you approach it from negative values.

I think there is a bit of discrepancy between rigid mathematical definitions and what is practically feasible in a numerical/computer implementation, so I would not over-specify this. I'm kind of fine with what we had originally, or something like
> a division by zero results in ±infinity or NaN, depending on the capabilities of the processing environment.

Comment (soxofaan) [reply divide.json:4]: I also wonder if we should recommend against (or forbid)  throwing a DivisionByZero error. If you're processing thousands/millions of pixels you probably prefer a NaN in some output pixels over your whole job failing.

Comment (m-mohr) [reply divide.json:4]: > I also wonder if we should recommend against (or forbid) throwing a DivisionByZero error. If you're processing thousands/millions of pixels you probably prefer a NaN in some output pixels over your whole job failing.

Well, the error only occurs if your environment doesn't support Nan/Infinity. It's difficult to return NaN if Nan is unsupported ;-)

Comment (soxofaan) [reply divide.json:4]: > It's difficult to return NaN if NaN is unsupported

sure, but we could recommend against throwing DivisionByZero if the environment *does* support NaN

File: log.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
     "id": "log",
     "summary": "Logarithm to a base",
-    "description": "Logarithm to the base `base` of the number `x` is defined to be the inverse function of taking b to the power of x.\n\nThe no-data value `null` is passed through and therefore gets propagated if any of the arguments is `null`.\n\nThe computations follow [IEEE Standard 754](https://ieeexplore.ieee.org/document/8766229) whenever the processing environment supports it. Therefore, `log(0, 2)` results in ±infinity if the processing environment supports it or otherwise an exception is thrown.",
+    "description": "Logarithm to the base `base` of the number `x` is defined to be the inverse function of taking b to the power of x.\n\nThe no-data value `null` is passed through and therefore gets propagated if any of the arguments is `null`.\n\nThe computations follow [IEEE Standard 754](https://ieeexplore.ieee.org/document/8766229) whenever the processing environment supports it. Therefore, having `x` set to `0` with any base results in -infinity if the processing environment supports it. Otherwise, an exception is thrown.",
```
Comment (soxofaan) [inline log.json:4]: We should probably  specify here (and for `ln`) that NaN is returned when x is negative too

**Comment by m-mohr** [comment]
I'll actually close here and merge this into #476 because the changes are overlapping.

File: log.json (hunk 2 above)
Comment (m-mohr) [reply log.json:4]: Should be part of #476 now.

**Pull request closed without merging**